src/components/Contexts/CollaboratorsContext.tsx

```diff
@@ -0,0 +1,334 @@
+   *
+   * @returns void
+   */
+  const handleAddCollaborator = (): void => {
+    setCurrentCollaborators((prev) => [...prev, createEmptyCollaborator()]);
+  };
+
+  /**
+   * Removes a collaborator from the list
+   * @param collaboratorIdx Index of the collaborator to remove
+   * @returns void
+   */
+  const handleRemoveCollaborator = (collaboratorIdx: number): void => {
+    const currentCollaborator: Collaborator = currentCollaborators[collaboratorIdx];
+
+    if (!currentCollaborator) {
+      return;
+    }
+
+    setCurrentCollaborators((prev) => {
+      const newCollaborators = prev.filter((_, index) => index !== collaboratorIdx);
+
+      if (!newCollaborators?.length) {
+        return [createEmptyCollaborator()];
+      }
+
+      return newCollaborators;
+    });
+  };
+
+  /**
+   * Updates a collaborator in the list
+   * @param collaboratorIdx Index of the collaborator to update
+   * @param newCollaborator New collaborator data
+   * @returns void
+   */
+  const handleUpdateCollaborator = (
+    collaboratorIdx: number,
+    newCollaborator: CollaboratorInput
+  ): void => {
+    if (
+      isNaN(collaboratorIdx) ||
+      (!newCollaborator?.collaboratorID && !newCollaborator?.permission)
+    ) {
+      return;
+    }
+
+    const potentialCollaborator = potentialCollaborators.find(
+      (pc) => pc.collaboratorID === newCollaborator?.collaboratorID
+    );
+
+    setCurrentCollaborators((prevCollaborators) => {
+      const collaboratorsClone = [...prevCollaborators];
+
+      collaboratorsClone[collaboratorIdx] = {
+        ...potentialCollaborator,
+        ...newCollaborator,
+      };
+
+      return collaboratorsClone;
+    });
+  };
+
+  /**
+   * Saves the current collaborators and returns the updated collaborators
+   * @returns {Promise<Collaborator[]>} Promise resolving to the updated list of collaborators
+   */
+  const saveCollaborators = useCallback(async (): Promise<Collaborator[]> => {
+    const collaboratorsToSave = currentCollaborators
+      .filter((c) => !!c.collaboratorID && !!c.permission)
+      .map((c) => ({ collaboratorID: c.collaboratorID, permission: c.permission }));
+
+    try {
+      const { data, errors } = await editSubmissionCollaborators({
+        variables: {
+          submissionID,
+          collaborators: collaboratorsToSave,
+        },
+      });
+
+      if (errors || !data?.editSubmissionCollaborators) {
+        throw new Error("Failed to save collaborators.");
+      }
+
+      enqueueSnackbar("All collaborator changes have been saved successfully.");
+
+      return data.editSubmissionCollaborators.collaborators;
+    } catch (err) {
+      Logger.error(`CollaboratorsContext: ${err.toString()}`);
+      enqueueSnackbar("Unable to edit submission collaborators.", {
+        variant: "error",
+      });
+      return [];
+    }
+  }, [currentCollaborators, editSubmissionCollaborators, enqueueSnackbar, submissionID]);
+
+  const contextValue = useMemo(
```

Comment:
Not a big deal but I would suggest adding the type here.

```suggestion
  const contextValue = useMemo<CollaboratorsCtxState>(
```

Or

```suggestion
  const contextValue: CollaboratorsCtxState = useMemo(
```

(Or both)